File: frontend-evoltis/src/app/services/services.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ServiceDto } from '../models/services/service.dto';
import { formatISO } from 'date-fns';
import { environment } from '../../environments/environment.development';
import { ServiceRequest } from '../models/services/service.request';
import { ServicesDto } from '../models/services/services.dto';

@Injectable({
  providedIn: 'root'
})
export class ServicesService {

  constructor(private http: HttpClient) { }

  getServices(
    aprox: string | null, 
    dateCreatedFrom: Date | null, 
    dateCreatedUntil: Date | null, 
    priceMin: number | null, 
    priceMax: number | null): Observable<ServicesDto>{

    let params = new URLSearchParams();
    if(aprox) params.set('Aprox', aprox);
    if(dateCreatedFrom) params.set('DateCreatedFrom', formatISO(dateCreatedFrom, { representation: 'complete' }));
    if(dateCreatedUntil) params.set('DateCreatedTo', formatISO(dateCreatedUntil, { representation: 'complete' }));
    if(priceMin) params.set('PriceMin', priceMin.toString());
    if(priceMax) params.set('PriceMax', priceMax.toString());

    return this.http.get<ServicesDto>(environment.apiUrl + `services?` + params.toString());
  }

  getServiceById(id: string): Observable<ServiceDto> {
    return this.http.get<ServiceDto>(environment.apiUrl + `services/${id}`);
  }

  postService(service: ServiceRequest): Observable<ServiceDto> {
    return this.http.post<ServiceDto>(environment.apiUrl + 'services', service);
  }

  modifyService(id: string, service: ServiceRequest): Observable<ServiceDto> {
    return this.http.put<ServiceDto>(environment.apiUrl + `services/${id}`, service);
  }

  deleteService(id: string): Observable<ServiceDto> {
    return this.http.delete<ServiceDto>(environment.apiUrl + `services/${id}`);
  }
}

File: backend-evoltis/backend-evoltis.CORE/Services/Imp/ServicesService.cs
using AutoMapper;
using backend_evoltis.CORE.Models.Services.Requests;
using backend_evoltis.DOMAIN.Entities;
using backend_evoltis.DOMAIN.Interfaces;

namespace backend_evoltis.CORE.Services.Imp
{
    public class ServicesService : IServicesService
    {
        private readonly IServiceRepository _repository;
        private readonly IMapper _mapper;

        public ServicesService(IServiceRepository repository, IMapper mapper)
        {
            _repository = repository;
            _mapper = mapper;
        }

        public async Task<Service> DeleteService(Guid id) => await _repository.DeleteService(id);

        public async Task<Service> GetServiceById(Guid id) => await _repository.GetServiceById(id);

        public async Task<List<Service>> GetServices(
            string? aprox, 
            DateTime? dateCreatedFrom, 
            DateTime? dateCreatedUntil, 
            decimal? priceMin, 
            decimal? priceMax) => await _repository.GetServices(
                aprox, 
                dateCreatedFrom, 
                dateCreatedUntil, 
                priceMin, 
                priceMax
                );

        public async Task<Service> ModifyService(Guid id, ServiceRequest request)
        {
            var service = await _repository.GetServiceById(id);
            service = _mapper.Map(request, service);
            service.ModifiedAt = DateTime.Now;
            return await _repository.ModifyService(service);
        }

        public async Task<Service> PostService(ServiceRequest request)
        {
            var service = _mapper.Map<Service>(request);
            service.ModifiedAt = DateTime.Now;
            return await _repository.PostService(service);
        }
    }
}
